ctru-rs/src/applets/mii_selector.rs

```
@@ -0,0 +1,193 @@
+            MiiConfigIndex::Index(i) => unsafe {
+                ctru_sys::miiSelectorBlacklistGuestMii(self.config.as_mut(), i)
+            },
+            MiiConfigIndex::All => unsafe {
+                ctru_sys::miiSelectorBlacklistGuestMii(
+                    self.config.as_mut(),
+                    ctru_sys::MIISELECTOR_GUESTMII_SLOTS,
+                )
+            },
+        }
+    }
+
+    pub fn whitelist_user_mii(&mut self, mii_index: MiiConfigIndex) {
+        match mii_index {
+            MiiConfigIndex::Index(i) => unsafe {
+                ctru_sys::miiSelectorWhitelistUserMii(self.config.as_mut(), i)
+            },
+            MiiConfigIndex::All => unsafe {
+                ctru_sys::miiSelectorWhitelistUserMii(
+                    self.config.as_mut(),
+                    ctru_sys::MIISELECTOR_USERMII_SLOTS,
+                )
+            },
+        }
+    }
+
+    pub fn blacklist_user_mii(&mut self, mii_index: MiiConfigIndex) {
+        match mii_index {
+            MiiConfigIndex::Index(i) => unsafe {
+                ctru_sys::miiSelectorBlacklistUserMii(self.config.as_mut(), i)
+            },
+            MiiConfigIndex::All => unsafe {
+                ctru_sys::miiSelectorBlacklistUserMii(
+                    self.config.as_mut(),
+                    ctru_sys::MIISELECTOR_USERMII_SLOTS,
+                )
+            },
+        }
+    }
+
+    // This function is static inline in libctru
+    // https://github.com/devkitPro/libctru/blob/af5321c78ee5c72a55b526fd2ed0d95ca1c05af9/libctru/include/3ds/applets/miiselector.h#L155
+    pub fn set_initial_index(&mut self, index: u32) {
+        self.config.initial_index = index
+    }
+
+    pub fn launch(&mut self) -> MiiSelectorReturn {
+        let mut return_val = Box::<ctru_sys::MiiSelectorReturn>::default();
+        unsafe { ctru_sys::miiSelectorLaunch(self.config.as_mut(), return_val.as_mut()) }
+
+        return_val.into()
+    }
+}
+
+impl MiiSelectorReturn {
```

These functions are duplicate features already present (name and author can be obtained by the `MiiData`).

Furthermore, the checksum (while useful in a C environment) is a case we just shouldn't expect the user to check. The case of a faulty checksum means that our `MiiSelectorReturn` wrapped value is invalid, which is one of the safety measures we should prevent *by default*.

The checksum should be checked before returning the mii data to the user, and if the check fails, it should return an error. This check should be handled in the `MiiSelector::launch` method.